src/pages/VideoDetails.jsx

```diff
@@ -0,0 +1,61 @@
+import { Link } from 'react-router-dom';
+import ReactPlayer from 'react-player';
+import SuggestedVideoCard from '../components/SuggestedVideoCard';
+import { VideosWrapper } from '../components/styles/VideosWrapper.styled';
+import { VideoContainer } from '../components/styles/VideoContainer.styled';
+import { SuggestedVideosContainer } from '../components/styles/SuggestedVideosContainer.styled';
+import Alert from 'react-bootstrap/Alert';
+import Spinner from '../components/Spinner';
+import PropTypes from 'prop-types';
+import { useParams } from 'react-router-dom';
+
+const VideoDetails = ({ loading, error, videoData }) => {
+  const youtubeUrl = 'https://www.youtube.com/watch?v=';
+  const { id } = useParams();
+  if (error) {
+    console.error(error);
+    return <Alert variant="danger">An error occured.</Alert>;
+  }
+  const videos = videoData ? videoData.items : [];
+  const [selectedVideo] = videos.filter((video) => video.id.videoId === id);
+  const videoTitle = selectedVideo ? selectedVideo.snippet.title : '';
+  const videoDescription = selectedVideo
+    ? selectedVideo.snippet.description
+    : '';
```

Comment:
instead of doing all of this try fetching the video info from the api directly. Try to search how you can do it but if you are unable to find it reach me out on slack